builder/src/package_metadata.rs

```diff
@@ -0,0 +1,197 @@
+//! Metadata to customize builds on docs.rs.
+
+use crate::Result;
+use cargo::core::Package;
+use failure::err_msg;
+use std::fs;
+use std::path::Path;
+use toml::Value;
+
+/// PackageMetadata for custom builds
+///
+/// You can customize docs.rs builds by defining `[package.metadata.docs.rs]` table in your
+/// crates' `Cargo.toml`.
+///
+/// An example metadata:
+///
+/// ```text
+/// [package]
+/// name = "test"
+///
+/// [package.metadata.docs.rs]
+/// features = [ "feature1", "feature2" ]
+/// all-features = true
+/// no-default-features = true
+/// default-target = "x86_64-unknown-linux-gnu"
+/// rustc-args = [ "--example-rustc-arg" ]
+/// rustdoc-args = [ "--example-rustdoc-arg" ]
+/// dependencies = [ "example-system-dependency" ]
+/// ```
+///
+/// You can define one or more fields in your `Cargo.toml`.
+pub struct PackageMetadata {
+    /// List of features docs.rs will build.
+    ///
+    /// By default, docs.rs will only build default features.
+    pub features: Option<Vec<String>>,
+
+    /// Set `all-features` to true if you want docs.rs to build all features for your crate
+    pub all_features: bool,
+
+    /// Docs.rs will always build default features.
+    ///
+    /// Set `no-default-fatures` to `false` if you want to build only certain features.
+    pub no_default_features: bool,
+
+    /// Docs.rs is running on `x86_64-unknown-linux-gnu` target system and default documentation
+    /// is always built on this target. You can change default target by setting this.
+    pub default_target: Option<String>,
+
+    /// List of command line arguments for `rustc`.
+    pub rustc_args: Option<Vec<String>>,
+
+    /// List of command line arguments for `rustdoc`.
+    pub rustdoc_args: Option<Vec<String>>,
+
+    /// System dependencies.
+    ///
+    /// Docs.rs is running on a Debian jessie.
```

Comment:
More importantly system dependencies is not being used anywhere. Maybe we can actually try to install them this time since `docsrs-builder` will be run in a disposable container.

Awesome catch!

Reply:
I added code to install system dependencies if it's running in our build environment.